framework/src/Volo.Abp.EntityFrameworkCore.PostgreSql/Volo/Abp/EntityFrameworkCore/DistributedEvents/PostgreSqlDbContextEventInbox.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Options;
+using Volo.Abp.EventBus.Distributed;
+using Volo.Abp.Timing;
+using Volo.Abp.Uow;
+
+namespace Volo.Abp.EntityFrameworkCore.DistributedEvents
+{
+    public class PostgreSqlDbContextEventInbox<TDbContext> : DbContextEventInbox<TDbContext>, IPostgreSqlDbContextEventInbox<TDbContext>
+        where TDbContext : IHasEventInbox
+    {
+        public PostgreSqlDbContextEventInbox(
+            IDbContextProvider<TDbContext> dbContextProvider,
+            IClock clock,
+            IOptions<AbpDistributedEventBusOptions> distributedEventsOptions)
+            : base(dbContextProvider, clock, distributedEventsOptions)
+        {
+        }
+
+        [UnitOfWork]
+        public override async Task MarkAsProcessedAsync(Guid id)
+        {
+            var dbContext = await DbContextProvider.GetDbContextAsync();
+            var tableName = dbContext.IncomingEvents.EntityType.GetSchemaQualifiedTableName();
+
+            var sql = $"UPDATE \"{tableName}\" SET \"Processed\" = '1', \"ProcessedTime\" = '{Clock.Now}' WHERE \"Id\" = '{id}'";
+            await dbContext.Database.ExecuteSqlRawAsync(sql);
+        }
+
+        [UnitOfWork]
+        public override async Task DeleteOldEventsAsync()
+        {
+            var dbContext = await DbContextProvider.GetDbContextAsync();
+            var tableName = dbContext.IncomingEvents.EntityType.GetSchemaQualifiedTableName();
+            var timeToKeepEvents = Clock.Now.Add(DistributedEventsOptions.InboxKeepEventTimeSpan);
```

Comment:
Should we move this option into `AbpEventBusBoxesOptions`? You've created a new options class, but this is added to `DistributedEventsOptions`.